dmc/projects/services/git_service/gitlab_git_service.rb

```diff
@@ -29,6 +29,12 @@ def commit(dir, sha)
       Gitlab::Git::Commit.find(repo(dir), sha)
     end
 
+    def pull(dir)
+      Dir.chdir(dir) do
```

Comment:
I'm not really big fan of this: moving to another directory, run a system command. What do you think to use `Rugged` for that? It's already in the `Gemfile,` so you could ask to `Rugged` to fetch the repository.